cpp/include/raft/linalg/eig.cuh

```diff
@@ -112,6 +114,117 @@ void eigJacobi(const raft::handle_t& handle,
 {
   detail::eigJacobi(handle, in, n_rows, n_cols, eig_vectors, eig_vals, stream, tol, sweeps);
 }
+
+/**
+ * @brief eig decomp with divide and conquer method for the column-major
+ * symmetric matrices
+ * @tparam ElementType the data-type of input and output
+ * @tparam IntegerType Integer used for addressing
+ * @param handle raft::handle_t
+ * @param in input raft::device_matrix_view (symmetric matrix that has real eig values and
+ * vectors)
+ * @param eig_vectors: eigenvectors output of type raft::device_matrix_view
+ * @param eig_vals: eigen values output of type raft::device_vector_view
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+void eig_dc(const raft::handle_t& handle,
+            const raft::device_matrix_view<ElementType, IndexType, raft::col_major> in,
```

Comment:
```suggestion
            raft::device_matrix_view<const ElementType, IndexType, raft::col_major> in,
```
Would you consider applying this change generally to the entire PR?

Also, would you consider renaming `ElementType` to `ValueType` here?  (Please see above notes.)